src/renderer/base/renderer.cpp

```diff
@@ -67,16 +69,11 @@ Renderer::~Renderer()
         return S_FALSE;
     }
 
-    for (IRenderEngine* const pEngine : _rgpEngines)
+    FOREACH_ENGINE(pEngine)
     {
         auto tries = maxRetriesForRenderEngine;
         while (tries > 0)
         {
-            if (_destructing)
-            {
-                return S_FALSE;
-            }

```

Comment:
Curious: moving this later ensures that we might try to paint a frame while destructing; why the change?

Reply:
Ah woops I did this from memory.